File: src/parser/twitch.py
from jinja2 import Template

TWITCH_FORMAT_TO_EXTENSION = {
    "static": "png",
    "animated": "gif",
}


def get_first(data, keys):
    for key in keys:
        if key in data:
            return data[key]

    return None


def parse_json(json_data):
    template = Template(json_data["template"])

    return_data = list(
        map(
            lambda x: {
                "id": x["id"],
                "name": x["name"],
                "format": x["format"][-1],
                "scale": x["scale"][-1],
                "theme_mode": x["theme_mode"][0],
            },
            json_data["data"],
        )
    )

    return_data = list(
        map(
            lambda x: {
                "name": x["name"],
                "extension": TWITCH_FORMAT_TO_EXTENSION[x["format"]],
                "url": template.render(x),
            },
            return_data,
        )
    )

    return return_data

File: src/parser/twitch.js
const Handlebars = require("handlebars");
const { format } = require("logform");

TWITCH_FORMAT_TO_EXTENSION = {
  static: "png",
  animated: "gif",
};

const getFirst = (dict, keys) => {
  for (const key of keys) {
    if (dict[key]) {
      return dict[key];
    }
  }
  return null;
};

const parseJson = (json) => {
  template = Handlebars.compile(json["template"]);

  const emotes = json["data"]
    .map((emote) => ({
      id: emote.id,
      name: emote["name"],
      format: emote["format"].at(-1),
      scale: emote.scale.at(-1),
      theme_mode: emote.theme_mode.at(0),
    }))
    .map((emote) => ({
      name: emote.name,
      extension: TWITCH_FORMAT_TO_EXTENSION[emote.format],
      url: template(emote),
    }));

  return emotes;
};

module.exports = { parseJson };
